### ISSUE TITLE
user-data: issue with gzip = true and base64_encode = false

### ISSUE THREAD

**Issue by mcorbin** [exoscale/terraform-provider-exoscale#207]
By making some tests with the provider (related to #205), I had this issue:

terraform file:

```
data "template_file" "cloud_init" {
  template = <<EOF
#cloud-config

write_files:
- path: /plop
  content: |
    plop
EOF
}


data "template_cloudinit_config" "user_data" {
  gzip          = true
  base64_encode = false

  part {
    filename     = "init.cfg"
    content_type = "text/cloud-config"
    content      = "${data.template_file.cloud_init.rendered}"
  }
}


variable "key_pair" {
  type    = "string"
  default = "dell exoscale"
}

variable "zone" {
  default = "ch-gva-2"
}

variable "static_machines" {
  default = 1
}

resource "exoscale_compute" "static" {
  user_data = "${data.template_cloudinit_config.user_data.rendered}"
  count = "${var.static_machines}"
  display_name = "demo-static-${count.index}"
  template = "Linux Debian 9 64-bit"
  size = "Small"
  disk_size = "10"
  security_groups = ["default", "ssh"]
  key_pair = "${var.key_pair}"
  zone = "${var.zone}"
}
```

terraform apply:

```
data.template_file.cloud_init: Refreshing state...
data.template_cloudinit_config.user_data: Refreshing state...

An execution plan has been generated and is shown below.
Resource actions are indicated with the following symbols:
  + create

Terraform will perform the following actions:

  + exoscale_compute.static
      id:                         <computed>
      affinity_group_ids.#:       <computed>
      affinity_groups.#:          <computed>
      disk_size:                  "10"
      display_name:               "demo-static-0"
      gateway:                    <computed>
      ip4:                        "true"
      ip6:                        "false"
      ip6_address:                <computed>
      ip6_cidr:                   <computed>
      ip_address:                 <computed>
      key_pair:                   "dell exoscale"
      name:                       <computed>
      password:                   <computed>
      security_group_ids.#:       <computed>
      security_groups.#:          "2"
      security_groups.3814588639: "default"
      security_groups.4002270276: "ssh"
      size:                       "Small"
      state:                      <computed>
      tags.%:                     <computed>
      template:                   "Linux Debian 9 64-bit"
      user_data:                  "\x1f\x8b\b\x00\x00\x00\x00\x00\x00\xffd\x8fAK\xf3@\x10\x86\xef\x03\xf9\x0fC\xbe\xf36\x9f'aK\x0fj{\xf0\x10\x05\xa9\x82'\xd9n&\xe9@vvٝ`\v\xfexiD\fz\x9c\xe1\x99y\x9f\xf7.\x8a\x92\xa8ٟ\x13Y\fӨ\x9c\\\xd6&\xf0\x89\xba5\x1e\xe2$\x9d\xcb\xe7M\xdd\u07b7\xbb\xdb\xc7\xe7\x87\xed\xcd\xd3k\r\x97ɼP.\x1c\xc5\xe2\xd5\xea\u007f\x05\x15\x18\xb3\x84*\xf8\xfe\xbd\xe5\x92ba\x9dY\xa7\xea\xfc1\x90\xe8\x1a{\x1eI\\\xa0M\xcdº\xf2\xfdP\xff\\\xed\xb3\x93\xd2S6;\xf1\xb1c\x19,^\x1fX\x17\xc0\xac\xact\xd2Əqꌏ\xd2\xf3PAˁ\xfe\xc8\xfd[2\x00\uf655\xde.\x02ł\xc1\xe4\xf4h\xb1IcL\x80\xe8\xbf\x12,~\x00\"\xe2\xbc\xfd]Ϙ\n>\x03\x00\x00\xff\xffݾ\xb2\x80=\x01\x00\x00"
      username:                   <computed>
      zone:                       "ch-gva-2"


Plan: 1 to add, 0 to change, 0 to destroy.

Do you want to perform these actions?
  Terraform will perform the actions described above.
  Only 'yes' will be accepted to approve.

  Enter a value: yes

exoscale_compute.static: Creating...
  affinity_group_ids.#:       "" => "<computed>"
  affinity_groups.#:          "" => "<computed>"
  disk_size:                  "" => "10"
  display_name:               "" => "demo-static-0"
  gateway:                    "" => "<computed>"
  ip4:                        "" => "true"
  ip6:                        "" => "false"
  ip6_address:                "" => "<computed>"
  ip6_cidr:                   "" => "<computed>"
  ip_address:                 "" => "<computed>"
  key_pair:                   "" => "dell exoscale"
  name:                       "" => "<computed>"
  password:                   "<sensitive>" => "<sensitive>"
  security_group_ids.#:       "" => "<computed>"
  security_groups.#:          "" => "2"
  security_groups.3814588639: "" => "default"
  security_groups.4002270276: "" => "ssh"
  size:                       "" => "Small"
  state:                      "" => "<computed>"
  tags.%:                     "" => "<computed>"
  template:                   "" => "Linux Debian 9 64-bit"
  user_data:                  "" => "\x1f\x8b\b\x00\x00\x00\x00\x00\x00\xffd\x8fAK\xf3@\x10\x86\xef\x03\xf9\x0fC\xbe\xf36\x9f'aK\x0fj{\xf0\x10\x05\xa9\x82'\xd9n&\xe9@vvٝ`\v\xfexiD\fz\x9c\xe1\x99y\x9f\xf7.\x8a\x92\xa8ٟ\x13Y\fӨ\x9c\\\xd6&\xf0\x89\xba5\x1e\xe2$\x9d\xcb\xe7M\xdd\u07b7\xbb\xdb\xc7\xe7\x87\xed\xcd\xd3k\r\x97ɼP.\x1c\xc5\xe2\xd5\xea\u007f\x05\x15\x18\xb3\x84*\xf8\xfe\xbd\xe5\x92ba\x9dY\xa7\xea\xfc1\x90\xe8\x1a{\x1eI\\\xa0M\xcdº\xf2\xfdP\xff\\\xed\xb3\x93\xd2S6;\xf1\xb1c\x19,^\x1fX\x17\xc0\xac\xact\xd2Əqꌏ\xd2\xf3PAˁ\xfe\xc8\xfd[2\x00\uf655\xde.\x02ł\xc1\xe4\xf4h\xb1IcL\x80\xe8\xbf\x12,~\x00\"\xe2\xbc\xfd]Ϙ\n>\x03\x00\x00\xff\xffݾ\xb2\x80=\x01\x00\x00"
  username:                   "" => "<computed>"
  zone:                       "" => "ch-gva-2"

Error: Error applying plan:

1 error(s) occurred:

* exoscale_compute.static: 1 error(s) occurred:

* exoscale_compute.static: API error Unauthorized 401 (CSErrorCode(0) 0): unable to verify user credentials and/or request signature

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.

```

The combinaison of `gzip = true` and `base64_encode = false` doesn't seem to work (I don't know if it's expected or not). The Cloudstack error is also strange (`API error Unauthorized`).

**Comment by greut** [comment]
> The combinaison of gzip = true and base64_encode = false doesn't seem to work

This is probably expected.

**Comment by greut** [comment]
Closed by 6aa525d7c1d4f39744149f73c87e865d3172f2b6

> Error: Error applying plan:
>
> 1 error(s) occurred:
> 
> * exoscale_compute.master: 1 error(s) occurred:
>
> * exoscale_compute.master: user_data appears gzipped. It should be left raw, or also be base64 encoded

**Comment by greut** [comment]
fixed by #206
